tests/test_freshdesk_bookmarks.py

```diff
@@ -0,0 +1,231 @@
+                first_sync_count = first_sync_record_count.get(stream, 0)
+                second_sync_count = second_sync_record_count.get(stream, 0)
+                first_sync_records = [record.get('data') for record in
+                                       first_sync_messages.get(stream).get('messages')
+                                       if record.get('action') == 'upsert']
+                second_sync_records = [record.get('data') for record in
+                                        second_sync_messages.get(stream).get('messages')
+                                        if record.get('action') == 'upsert']
+                if stream != {'conversations'}:  # conversations has no bookmark
+                    first_bookmark_value = first_sync_bookmarks.get(stream)
+                    second_bookmark_value = second_sync_bookmarks.get(stream)
+
+                if expected_replication_method == self.INCREMENTAL:
+
+                    # collect information specific to incremental streams from syncs 1 & 2
+                    replication_key = next(iter(expected_replication_keys[stream]))
+                    if stream != {'conversations'}:  # conversations have no bookmark
+                        simulated_bookmark_value = simulated_states[stream]
+
+                    if stream == {'conversations'}:
+                        print("*** Only checking sync counts for conversations stream ***")
+                        # TODO discuss re-factor to use tickets bookmark for conversations assertions
+                        # Verify the number of records in the 2nd sync is less then the first
+                        self.assertLessEqual(second_sync_count, first_sync_count)
+                        if second_sync_count == first_sync_count:
+                            print("WARN: first_sync_count == second_sync_count for stream: {}".format(stream))
+
+                        continue
+
+                    # Verify the first sync sets a bookmark of the expected form
+                    self.assertIsNotNone(first_bookmark_value)
+
+                    # Verify the second sync sets a bookmark of the expected form
+                    self.assertIsNotNone(second_bookmark_value)
+
+                    # Verify the second sync bookmark is Equal to the first sync bookmark
+                    # assumes no changes to data during test
+                    self.assertEqual(second_bookmark_value, first_bookmark_value)
+
+                    # Verify the number of records in the 2nd sync is less then the first
+                    self.assertLessEqual(second_sync_count, first_sync_count)
+                    if second_sync_count == first_sync_count:
+                        print("WARN: first_sync_count == second_sync_count for stream: {}".format(stream))
+
+                    # Verify the bookmark is the max value sent to the target for a given replication key.
+                    rec_time = []
```

Comment:
TODO: Is it bad to sort these ourselves?

Reply:
Keeping sort for now even though it may not do anything if data is sent pre-sorted from tap.  But the pre-sort is not 100% guaranteed.